app/src/main/java/br/com/stonesdk/sdkdemo/activities/MifareActivity.kt

```diff
@@ -0,0 +1,323 @@
+     * @param sector Número do setor.
+     * @param block Número do bloco relativo ao setor (mínimo 0, máximo 3).
+     * @param key Chave de autenticação do bloco.
+     */
+    private fun executeBlockRead(sector: Int, block: Int, key: ByteArray) {
+        val mifareProvider = PosMifareProvider(applicationContext)
+        mifareProvider.connectionCallback = object : StoneCallbackInterface {
+            override fun onSuccess() {
+                setCardUUIDText(mifareProvider.cardUUID.contentToString())
+                logTextView!!.append("Cartão detectado: " + mifareProvider.cardUUID.contentToString() + "\n")
+
+                // Autentica o setor
+                try {
+                    mifareProvider.authenticateSector(MifareKeyType.TypeA, key, sector.toByte())
+                } catch (e: PosMifareProvider.MifareException) {
+                    Toast.makeText(this@MifareActivity, "Erro na autenticação", Toast.LENGTH_SHORT)
+                        .show()
+                    logTextView!!.append(mifareProvider.listOfErrors.toString() + "\n")
+                    return
+                }
+
+                // Lê o valor de um bloco no setor
+                // O valor lido será escrito no byteArray caso a leitura ocorra com sucesso
+                val byteArray = ByteArray(16)
+                try {
+                    mifareProvider.readBlock(sector.toByte(), block.toByte(), byteArray)
+                } catch (e: PosMifareProvider.MifareException) {
+                    Toast.makeText(
+                        this@MifareActivity,
+                        "Erro na leitura do bloco",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                    logTextView!!.append(e.errorEnum.name + "\n")
+                    return
+                }
+
+                Toast.makeText(
+                    this@MifareActivity,
+                    "Valor lido: " + byteArray.contentToString(),
+                    Toast.LENGTH_SHORT
+                ).show()
+                logTextView!!.append("Valor lido: " + String(byteArray) + "\n")
+            }
+
+            override fun onError() {
+                runOnUiThread {
+                    Toast.makeText(this@MifareActivity, "Erro na detecção", Toast.LENGTH_SHORT)
+                        .show()
+                    logTextView!!.append(mifareProvider.listOfErrors.toString() + "\n")
+                }
+            }
+        }
+
+        mifareProvider.execute()
+    }
+
+    /**
+     * Exemplo de escrita em um bloco.
+     * @param sector Número do setor.
+     * @param block Número do bloco relativo ao setor (mínimo 0, máximo 3).
+     * @param key Chave de autenticação do bloco.
+     * @param value Valor que será gravado no bloco. (byte[16]).
+     */
+    private fun executeBlockWrite(sector: Int, block: Int, key: ByteArray, value: ByteArray) {
+        val mifareProvider = PosMifareProvider(applicationContext)
```

Comment:
Mesma coisa do executeBlockRead.
```suggestion
        mifareProvider = PosMifareProvider(applicationContext)
```

Reply:
Commitar a sugestão no github vai quebrar o build. Precisa alterar mais coisa pra lidar com o mifareProvider nullable.